src/Frontend/src/components/failedmessages/MessageView.vue

```diff
@@ -123,14 +122,15 @@ async function downloadHeadersAndBody(message: ExtendedFailedMessage) {
   try {
     const [, data] = await useTypedFetchFromServiceControl<Message[]>(`messages/search/${message.message_id}`);
 
-    if (data[0] == null) {
+    const messageDetails = data.find((value) => value.receiving_endpoint.name === message.receiving_endpoint?.name);
```

Comment:
This is the actual fix, instead of just taking the first message, we pick the correct one based on `receiving_endpoint` address